### PR TITLE
Support Django 2.0

### PR DESCRIPTION
**Pull request by blueyed**
Includes https://github.com/pytest-dev/pytest-django/pull/526.

### CODE REVIEW

**Comment by blueyed** [comment]
@pelme 
Would need some input/help with the private attributes you have used in https://travis-ci.org/pytest-dev/pytest-django/jobs/284386224#L564.

**Comment by blueyed** [comment]
@auvipy 
Thanks for the "review", but it was clearly not testing dj20, and is currently broken in ~3 places.
Maybe you have some experience/insight there?

**Comment by pelme** [comment]
@blueyed Thanks for working on this. Looking forward to Django 2.0!

https://github.com/django/django/commit/fba4f831bc75369c975a95c9b7774e9e89f8a2f9#diff-e0a696d19bf764562a439d3080a30fda seems to be the cause here. We should look for `apps.loading` instead of `apps._lock.locked()` which is nicer. :)

This would probably do the trick (disclaimer: I have not actually run this code):
```py
def apps_is_loading():
    if hasattr(apps, 'loading'):
        return apps.loading
    return apps._lock.locked()
```

File: tox.ini [hunk 1]
```diff
@@ -18,12 +15,13 @@ deps =
...
-    django1.9: Django>=1.9,<1.10
-    django1.8: Django>=1.8,<1.9
-    django1.7: Django>=1.7,<1.8
+    djmaster: https://github.com/django/django/archive/master.tar.gz
+    dj20: Django>=2.0a1,<2.1
+    dj111: Django>=1.11,<1.12
+    dj110: Django>=1.10,<1.11
+    dj19: Django>=1.9,<1.10
+    dj18: Django>=1.8,<1.9
+    dj17: Django>=1.7,<1.8
```
Comment (auvipy) [inline tox.ini]: django 1.7 should be removed

**Comment by auvipy** [comment]
@blueyed you could allow failures for 2.0 version in travis matrix and fix the 2.x errors in a separate pr. I saw old url patterns being used in one place

**Comment by michael-k** [comment]
I fixed the middleware issues: blueyed/pytest-django#2

File: tox.ini (hunk 1 above)
Comment (michael-k) [reply tox.ini]: I opened a separate PR for that: pytest-dev/pytest-django#524

**Comment by blueyed** [comment]
@pelme 
Thanks.  `apps.loading` will not become `False` after loading finished however - adjusted the test.

**Comment by blueyed** [comment]
@auvipy 
Sure, but I want to keep this out of this PR to not mix too many things.

**Comment by blueyed** [comment]
@auvipy @pelme 
Good to go?

**Comment by adamchainz** [comment]
Yo can we get a PyPI release with this? I test all my projects with `pytest-django` so can't make them Django 2 ready until this is out 😁 😁

**Comment by blueyed** [comment]
@adamchainz 
Sorry, I guess you have to use the master tarball then for now.
As for us using the latest pytest release seems to work with testing against dj20 however - so what is the problem you are seeing exactly?

**Comment by adamchainz** [comment]
My bad, I misread a few stack traces - the problems were really in my library code :)

**Pull request merged**